File: COMP1005/a1/a1.rb
################
# Assignment 1 #
################



##############
# Question 1 #
##############
def f2c(t)
	return (t - 32) * 5.fdiv(9)
end
#tests
puts "f2c #{f2c(80)}"
puts "f2c #{f2c(100)}"
puts "f2c #{f2c(300)}"

##############
# Question 2 #
##############
def lboz2kg(p,o=0)
	return ((p * 16) + o).fdiv(35.274)
end
#tests
puts "lboz2kg #{lboz2kg(2.2047)}"
puts "lboz2kg #{lboz2kg(50, 80)}"
puts "lboz2kg #{lboz2kg(10, 100)}"

##############
# Question 3 #
##############
def in2ftin(h)
	if h < 12
		return [0, h]
	else
		return [h / 12, ((h.fdiv(12)) - (h / 12)) * 10]
	end
end
#tests
puts "in2ftin #{in2ftin(12)}"
puts "in2ftin #{in2ftin(120)}"
puts "in2ftin #{in2ftin(150)}"
puts "in2ftin #{in2ftin(235)}"
puts "in2ftin #{in2ftin(5)}"

##############
# Question 4 #
##############
def bibformat_mla(author, title, city, publisher, year)
	return "#{author}. #{title}. #{city}: #{publisher}, #{year}"
end
#tests
puts "bibformat_mla #{bibformat_mla("Randall Munroe", "what if?", "New York", "Houghton Mifflin Hardcourt", 2014)}"
puts "bibformat_mla #{bibformat_mla("J. R. R. Tolkien", "The Hobbit", "London", "Unwin Paperbacks", "1981")}"

##############
# Question 5 #
##############
def bibformat_apa(author, title, city, publisher, year)
	return "#{author} (#{year}). #{title}. #{city}: #{publisher}."
end
#tests
puts "bibformat_apa #{bibformat_apa("Randall Munroe", "what if?", "New York", "Houghton Mifflin Hardcourt", 2014)}"
puts "bibformat_apa #{bibformat_apa("J. R. R. Tolkien", "The Hobbit", "London", "Unwin Paperbacks", "1981")}"

##############
# Question 6 #
##############
def bmi(w, h)
	return (w.fdiv(h * h)) * 703
end
#tests
puts "bmi #{bmi(180, 70)}"
puts "bmi #{bmi(500, 200)}"
puts "bmi #{bmi(99, 80)}"

##############
# Question 7 #
##############
def bmi_calculator()
	print "First name: " 
	name = gets.chomp
	print "Enter weight: "
	weight = gets.chomp.to_i
	print "Enter height: "
	height = gets.chomp.to_i
	puts "Looks like #{name}'s BMI is #{bmi(weight, height)}"
end
bmi_calculator()

File: COMP1005/a1/a1.cpp
#include <iostream>
#include <vector>
#include <string>
#include <algorithm>

using namespace std;

float f2c(int t){
	return (t - 32) * 5 / 9;
}

float lboz2kg(int p, int o){
	return ((p * 16) + o) / 35.274;
}

vector<int> in2ftin(int h){
	if (h < 12){
		return {0, h};
	} else {
		return {h / 12, ((h / 12) - (h / 12)) * 10};
	}
}

string bibformat_mla(string author, string title, string city, string publisher, string year){
	return author + ". " + title + ". " + city + ": " + publisher + ", " + year;
}

string bibformat_apa(string author, string title, string city, string publisher, string year){
	return author + " (" + year + "). " + title + ". " + city + ": " + publisher + ".";
}

float bmi(int w, int h){
	return (w / (h * h)) * 703;
}

void bmi_calculator(){
	string name;
	int weight, height;
	cout << "Name:";
	cin >> name;
	cout << "Enter weight:";
	cin >> weight;
	cout << "Enter height:";
	cin >> height;
	cout << "Looks like " << name << "'s BMI is "<< bmi(weight, height) << endl;
}

int main(int argc, char const *argv[]){
	cout << f2c(80) << endl;
	cout << lboz2kg(50, 80) << endl;
	cout << in2ftin(120)[0] << ", " << in2ftin(120)[1] << endl;
	cout << bibformat_mla("Randall Munroe", "what if?", "New York", "Houghton Mifflin Hardcourt", "2014") << endl;
	cout << bibformat_apa("Randall Munroe", "what if?", "New York", "Houghton Mifflin Hardcourt", "2014") << endl;
	cout << bmi(180, 70) << endl;
	bmi_calculator();
	return 0;
}